src/blocks/battery.rs

```diff
@@ -167,10 +167,19 @@ impl BatteryDevice for PowerSupplyDevice {
 
         let power_path = self.device_path.join("power_now");
         let current_path = self.device_path.join("current_now");
+        let voltage_path = self.device_path.join("voltage_now");
         let usage = if power_path.exists() {
             read_file("battery", &power_path)?
                 .parse::<f64>()
                 .block_error("battery", "failed to parse power_now")?
+        } else if current_path.exists() && voltage_path.exists() {
+            let current = read_file("battery", &current_path)?
+                .parse::<f64>()
+                .block_error("battery", "failed to parse current_now")?;
+            let voltage = read_file("battery", &voltage_path)?
+                .parse::<f64>()
+                .block_error("battery", "failed to parse voltage_now")?;
+            current * voltage
         } else if current_path.exists() {
             read_file("battery", &current_path)?
                 .parse::<f64>()
```

Comment:
This is the part i would suggest to drop because it breaks ERT calculation on Amp-based device drivers:

```suggestion
        let power_path = self.device_path.join("power_now");
        let current_path = self.device_path.join("current_now");
        let usage = if power_path.exists() {
            read_file("battery", &power_path)?
                .parse::<f64>()
                .block_error("battery", "failed to parse power_now")?
        } else if current_path.exists() {
            read_file("battery", &current_path)?
                .parse::<f64>()
```